src/VIPServer.java

```diff
@@ -139,10 +121,27 @@ public void main(String[] args) throws HostFailureException {
 					if (VIPSimulator.numberOfMergeJobs == 0)
 						stop = true;
 					if (runningMergeWorkers < VIPSimulator.numberOfMergeJobs) {
-						Msg.info("All GATE workers sent a 'GATE_END' message. Wake up " + mergeWorkers.size()
-								+ " Merge worker(s)");
-						runningMergeWorkers++;
-						mergeWorkers.firstElement().begin();
+
+						new Process(this.getHost(), "Timer") {
+							public void main(String[] args) throws HostFailureException {
+								Process.sleep(VIPSimulator.sosTime);
+								boolean mergeFlag = false;
+								while (runningMergeWorkers < VIPSimulator.numberOfMergeJobs) {							
+									while(!mergeFlag){
+										if(mergeWorkers.size() > 0){
+											mergeWorkers.get(runningMergeWorkers).begin();
```

Comment:
This will start the first merger in the vector (possibly several times), not all of them